fili-sql/src/main/java/com/yahoo/bard/webservice/sql/presto/DruidQueryToPrestoConverter.java

```diff
@@ -0,0 +1,296 @@
+     * Difference between this and the base class is the special column lower underscore to lower camel case conversion
+     * for Druid column names
+     * @param builder  The RelBuilder created with Calcite.
+     * @param druidQuery  The druid query to get the aggregations of.
+     * @param apiToFieldMapper  The mapping from api to physical name.
+     *
+     * @return the list of aggregations.
+     */
+    @Override
+    protected List<RelBuilder.AggCall> getAllQueryAggregations(
+            RelBuilder builder,
+            DruidAggregationQuery<?> druidQuery,
+            ApiToFieldMapper apiToFieldMapper
+    ) {
+        return druidQuery.getAggregations()
+                .stream()
+                .map(aggregation -> druidSqlAggregationConverter.apply(aggregation, apiToFieldMapper))
+                .map(optionalSqlAggregation -> optionalSqlAggregation.orElseThrow(() -> {
+                    String msg = "Couldn't build sql aggregation with " + optionalSqlAggregation;
+                    LOG.debug(msg);
+                    return new RuntimeException(msg);
+                }))
+                .map(sqlAggregation -> builder.aggregateCall(
+                        sqlAggregation.getSqlAggFunction(),
+                        false,
+                        null,
+                        CaseFormat.LOWER_UNDERSCORE.to(
+                                CaseFormat.LOWER_CAMEL,
+                                sqlAggregation.getSqlAggregationAsName()
+                        ),
+                        builder.field(sqlAggregation.getSqlAggregationFieldName())
+                ))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the post-aggregations of the query.
+     *
+     * @param builder the RelBuilder
+     * @param druidQuery the source druid query
+     * @param apiToFieldMapper api column to logic column name mapping
+     * @return a list of RexNode representing the post aggregation
+     */
+    private List<RexNode> getPostAggregations(
+            RelBuilder builder,
+            DruidAggregationQuery<?> druidQuery,
+            ApiToFieldMapper apiToFieldMapper
+    ) {
+        DruidPostAggregationToPresto druidPostAggregationToSql = new DruidPostAggregationToPresto();
+        List<RexNode> postAggregationFields = new ArrayList<>();
+        druidQuery.getPostAggregations().stream()
+                .map(postAggregation -> druidPostAggregationToSql
+                        .evaluatePostAggregation(postAggregation, builder, apiToFieldMapper))
+                .forEach(postAggregationFields::add);
+        return postAggregationFields;
+    }
+
+    /**
+     * Determines whether or not a query is able to be processed using
+     * the Sql backend.
+     *
+     * @param druidQuery  The query to check if is able to be processed.
+     *
+     * @return true if a valid query, else false.
+     */
+    protected boolean isValidQuery(DruidQuery<?> druidQuery) {
```

Comment:
remove?

Reply:
Cannot remove unless making it public in parent class.